test/unit/app-lifecycle.spec.js

```diff
@@ -99,6 +99,44 @@ describe('App-Lifecycle', function() {
     });
   });
 
+  describe('when restarting the app', function() {
+    beforeEach(function() {
+      this.beforeStartSpy = sinon.spy(function(options) {
```

Comment:
I think this is a good place for a stub instead of a spy
Then you can can use calledWith instead of returning a value